src/ARCtrl/ARCtrl.fs

```diff
@@ -242,7 +242,108 @@ type ARC(?isa : ISA.ArcInvestigation, ?cwl : CWL.CWL, ?fs : FileSystem.FileSyste
         let fsCopy = _fs.Copy()
         new ARC(?isa = isaCopy, ?cwl = _cwl, fs = fsCopy)
         
+    /// <summary>
+    /// Returns the FileSystemTree of the ARC with only the registered files and folders included.
+    /// </summary>
+    /// <param name="IgnoreHidden">Wether or not to ignore hidden files and folders starting with '.'. If true, no hidden files are included in the result. (default: true)</param>
+    member this.GetRegisteredPayload(?IgnoreHidden:bool) =
+
+        let isaCopy = _isa |> Option.map (fun i -> i.Copy()) // not sure if needed, but let's be safe
```

Comment:
As far as I see it, this is not needed. But we definitely need structural equality functions @Freymaurer. So functions like this here can be tested for doing no change to the ARC itself.

Reply:
then i guess it is better to leave it for now